oak_runtime/src/node/grpc/server/mod.rs

```diff
@@ -402,25 +401,22 @@ impl ServerStreamingService<Vec<u8>> for GrpcInvocationHandler {
             let response_iter = handler
                 .inject_grpc_request(grpc_request, &oak_label)
                 .map_err(|()| tonic::Status::new(tonic::Code::Internal, ""))?;
-
-            // First convert the `Iterator<Item = GrpcResponse>` to an
-            // `Iterator<Item = Result<Vec<u8>, tonic::Status>`.
-            let result_iter = response_iter.map(|response| {
-                debug!("Returning gRPC response: {:?}", response);
-                match response.status {
-                    None => Ok(response.rsp_msg),
-                    Some(status) if status.code == rpc::Code::Ok as i32 => Ok(response.rsp_msg),
-                    Some(status) => Err(to_tonic_status(status)),
+            let (tx, rx) = mpsc::unbounded_channel();
+
+            // Spawn a separate thread to handle responses, because the underlying call to
+            // Runtime::wait_on_channels() blocks.
```

Comment:
This feels not ideal since it's mixing threads and futures, which is not normally expected (though I understand why it's necessary at this point). How / when do we expect to clean this up? Will it require the async runtime that we have been talking about with @wildarch ? Perhaps worth adding a TODO here and refer to either an existing or a new issue.

Reply:
I don't think this is directly related to #186, that only concerns the SDK. The work there just builds on the existing synchronous ABI we have for nodes.

Is there a bug somewhere to make `Runtime::wait_on_channels` async?

Reply:
There is now: #1376